apache/superset
# refactor: New Icon system with Enhanced Antd Custom Icon

### SUMMARY
This PR introduces an entirely new Icon system that uses a combination of the Antd Custom Icon and Emotion while consolidating the usage of icons.

- Use existing Antd icons with the ability to enhance them
- Use custom icons trough the Antd Custom Icon component and the ability to enhance them
- Adds the ability to override any Antd existing icon with a custom one

As an example implementation, the following sections have been changed to use some of the icons under the new icon system:

- Charts lists
- Dashboards lists
- Saved Queries lists
- Datasets lists
- Databases lists
- Explore - Edit Datasource columns

## BEFORE

<img width="1580" alt="charts-list-BEFORE" src="https://user-images.githubusercontent.com/60598000/105756142-847efc80-5f4c-11eb-9ef7-f208feff4467.png">
<img width="1478" alt="dashboard-list-trash-BEFORE" src="https://user-images.githubusercontent.com/60598000/105756148-85b02980-5f4c-11eb-8cbf-78db689203f3.png">
<img width="1113" alt="dashboard-lists-cardview-BEFORE" src="https://user-images.githubusercontent.com/60598000/105756149-8648c000-5f4c-11eb-9656-f0b00ffa3aa3.png">
<img width="1394" alt="databases-lists-BEFORE" src="https://user-images.githubusercontent.com/60598000/105756152-86e15680-5f4c-11eb-89e6-b85312362ca0.png">
<img width="1455" alt="datasets-list-BEFORE" src="https://user-images.githubusercontent.com/60598000/105756154-8779ed00-5f4c-11eb-9e81-8259cae6a2b7.png">
<img width="1330" alt="saved-query-list-trash-BEFORE" src="https://user-images.githubusercontent.com/60598000/105756155-8779ed00-5f4c-11eb-895d-8086b3e4aae1.png">
<img width="871" alt="Screen Shot 2021-02-03 at 17 14 41" src="https://user-images.githubusercontent.com/60598000/106776714-9f382c00-6644-11eb-9d98-58a35fc945d9.png">


## AFTER

<img width="1571" alt="charts-list-AFTER" src="https://user-images.githubusercontent.com/60598000/105756179-91035500-5f4c-11eb-8d61-97a2ae0910ef.png">
<img width="1653" alt="dashboard-list-trash-AFTER" src="https://user-images.githubusercontent.com/60598000/105756183-919beb80-5f4c-11eb-99af-7328462f3675.png">
<img width="1107" alt="dashboard-lists-cardview-AFTER" src="https://user-images.githubusercontent.com/60598000/105756184-92348200-5f4c-11eb-9ffe-ab584db67a68.png">
<img width="1394" alt="databases-lists-AFTER" src="https://user-images.githubusercontent.com/60598000/105756188-92cd1880-5f4c-11eb-8fa3-209999a271c8.png">
<img width="1454" alt="datasets-list-AFTER" src="https://user-images.githubusercontent.com/60598000/105756190-9365af00-5f4c-11eb-9249-fd54df315879.png">
<img width="1327" alt="saved-query-list-trash-AFTER" src="https://user-images.githubusercontent.com/60598000/105756193-9365af00-5f4c-11eb-814e-cd02d18925ae.png">
<img width="867" alt="Screen Shot 2021-02-03 at 20 56 09" src="https://user-images.githubusercontent.com/60598000/106801652-4d9e9a00-6662-11eb-93de-5d57e0f23077.png">


### TEST PLAN
1. Go to any list
2. Hover your mouse on a list item
3. Make sure the trash icon appears and works as intended

### ADDITIONAL INFORMATION
